src/v/cloud_topics/level_one/metastore/frontend.cc

```diff
@@ -0,0 +1,413 @@
+
+namespace {
+
+ss::future<rpc::add_objects_reply> do_add_objects(
+  cluster::partition_manager&, model::ntp, rpc::add_objects_request) {
+    co_return rpc::add_objects_reply{.ec = rpc::errc::ok};
+}
+
+ss::future<rpc::replace_objects_reply> do_replace_objects(
+  cluster::partition_manager&, model::ntp, rpc::replace_objects_request) {
+    co_return rpc::replace_objects_reply{.ec = rpc::errc::ok};
+}
+
+ss::future<rpc::get_first_offset_ge_reply> do_get_first_offset_ge(
+  cluster::partition_manager&, model::ntp, rpc::get_first_offset_ge_request) {
+    co_return rpc::get_first_offset_ge_reply{.ec = rpc::errc::timed_out};
+}
+
+ss::future<rpc::get_first_timestamp_ge_reply> do_get_first_timestamp_ge(
+  cluster::partition_manager&,
+  model::ntp,
+  rpc::get_first_timestamp_ge_request) {
+    co_return rpc::get_first_timestamp_ge_reply{.ec = rpc::errc::timed_out};
+}
+
+ss::future<rpc::get_offsets_reply> do_get_offsets(
+  cluster::partition_manager&, model::ntp, rpc::get_offsets_request) {
+    co_return rpc::get_offsets_reply{.ec = rpc::errc::timed_out};
+}
+
+ss::future<rpc::get_compaction_offsets_reply> do_get_compaction_offsets(
+  cluster::partition_manager&,
+  model::ntp,
+  rpc::get_compaction_offsets_request) {
+    co_return rpc::get_compaction_offsets_reply{.ec = rpc::errc::timed_out};
+}
+
+} // namespace
+
+template<auto Func, typename req_t>
+requires requires(frontend::proto_t f, req_t req, ::rpc::client_opts opts) {
+    (f.*Func)(std::move(req), std::move(opts));
+}
+auto frontend::remote_dispatch(req_t request, model::node_id leader_id) {
+    using resp_t = req_t::resp_t;
+    return _connection_cache->local()
+      .with_node_client<proto_t>(
+        _self,
+        ss::this_shard_id(),
+        leader_id,
+        rpc_timeout,
+        [request = std::move(request)](proto_t proto) mutable {
+            return (proto.*Func)(
+              std::move(request),
+              ::rpc::client_opts{model::timeout_clock::now() + rpc_timeout});
+        })
+      .then(&::rpc::get_ctx_data<resp_t>)
+      .then([leader_id, self = _self](result<resp_t> r) mutable {
+          if (r.has_error()) {
+              vlog(
+                cd_log.warn,
+                "got error {} sending to L1 STM leader {} from node {}",
+                r.error().message(),
+                leader_id,
+                self);
+              return resp_t{.ec = rpc::errc::timed_out};
+          }
+          return std::move(r.value());
+      });
+}
+
+template<auto LocalFunc, auto RemoteFunc, typename req_t>
+requires requires(
+  experimental::cloud_topics::l1::frontend f,
+  const model::ntp& ntp,
+  req_t req) {
+    (f.*LocalFunc)(std::move(req), ntp, ss::shard_id{0});
+    request_has_metastore_partition<req_t>
+      || request_has_topic_id_partition<req_t>;
+}
+auto frontend::process(req_t req, bool local_only) {
+    using resp_t = req_t::resp_t;
+    return ensure_topic_exists().then([req = std::move(req), local_only, this](
```

Comment:
No coro love? :)

But we really should make this process simpler... 🤔

Reply:
Done